### PR TITLE
auto deploy react storybook on successful staging build [skip ui]

### PR DESCRIPTION
**Pull request by pcardune**

### CODE REVIEW

File: apps/script/deploy-storybook.sh [hunk 1]
```diff
@@ -11,8 +11,11 @@ SHA=`git rev-parse --verify HEAD`
 rm -rf $DIR_TO_DEPLOY
 
 # fetch version of application.css that is generated by rails
-echo "Fetching application.css from localhost, I hope you have dashboard running"
-curl -o build/package/css/application.css http://localhost-studio.code.org:3000/assets/application.css
+echo "Compiling application.css using rails asset pipeline"
+pushd ../dashboard
+rails assets:precompile || true
+cp public/assets/css/application-*.css ../apps/build/package/application.css
+popd
```
Comment (islemaster) [inline apps/script/deploy-storybook.sh:24]: neat!

File: circle.yml [hunk 2]
```diff
@@ -110,3 +110,9 @@ experimental:
     branches:
       only:
         - staging
+deployment:
+  storybook:
+    branch: staging
+    commands:
+      - 'echo "sync_assets: false" >> locals.yml'
+      - 'cd apps && npm run storybook:deploy'
```
Comment (islemaster) [inline circle.yml:118]: Does Circle have whatever credentials it needs to deploy this?

Comment (islemaster) [reply circle.yml:118]: Can we/do we implicitly skip this step if the storybook content hasn't changed?

File: apps/script/deploy-storybook.sh [hunk 3]
```diff
@@ -46,7 +46,7 @@ if [ -z "$CHANGED_FILES" ]; then
 fi
 
 # Commit the changes to the gh-pages branch
-git config user.name "Travis CI"
+git config user.name "Circle CI"
 git config user.email "paul@code.org"
```
Comment (islemaster) [inline apps/script/deploy-storybook.sh]: Seems like a bad thing that we've got your email address hard-coded here for a deploy step.

Comment (pcardune) [reply apps/script/deploy-storybook.sh]: yeah, what email should I use?

File: circle.yml (hunk 2 above)
Comment (pcardune) [reply circle.yml:118]: maybe? I don't think so. Need to figure out a good way to test that. Perhaps I should add this branch to the config and have it tested that way? We probably need to generate a token specifically for this repo.

File: apps/script/deploy-storybook.sh (hunk 3 above)
Comment (islemaster) [reply apps/script/deploy-storybook.sh]: How about dev @ ? We use that for our npm publish process.

File: circle.yml (hunk 2 above)
Comment (islemaster) [reply circle.yml:118]: Yeah, I'd temporarily add this branch to the config and try the deployment from here a few times.

**Pull request merged**